### ISSUE TITLE
Micromamba based setup?

### ISSUE THREAD

**Issue by jaimergp** [conda-incubator/setup-miniconda#75]
Thoughts on this? https://github.com/mamba-org/mamba#micromamba

> micromamba is a tiny version of the mamba package manager. It is a pure C++ package with a separate command line interface. It can be used to bootstrap environments (as an alternative to miniconda), but it's currently experimental. The benefit is that it's very tiny and does not come with a default version of Python.

It might be too soon because it's still experimental, but it could be helpful in the future.

**Comment by goanpeca** [comment]
Hi @jaimergp 

See the discussion on #66

**Comment by mariusvniekerk** [comment]
So ensureconda (https://github.com/conda-incubator/ensureconda) is a tiny little tool that like pip ensures that a conda (or conda-like) is around.  

It has some built in ways to install the standalone conda and micromamba.  Generally not intended as a replacement for things like miniconda but more for conda related tools like `conda-lock`.  Might suit some of your needs?

**Comment by mariusvniekerk** [comment]
We can certainly integrate some of the features from ensureconda into setup-miniconda if it makes sense

**Comment by goanpeca** [comment]
Hi @mariusvniekerk thanks for it. I have been looking at ensure conda for some work I will be doing in constructor, so thanks for that. 

The ideal place would be to have micromamba bundled with the github runners. Otherwise downloading things would be almost the same for the bundled miniconda (installing mamba) then downloading micromamba.

But to have things added to github runners (besides enough requests) is project maturity I think ?

**Comment by mariusvniekerk** [comment]
Yeah main difference is mostly the micromamba standalone installation doesn't need conda at all, so you save a conda solve for mamba.

**Comment by hadim** [comment]
The "Setup Conda" step when used with `mamba` seems to take a long time (~6 min). A `mamba` only setup would probably fix this.

**Comment by goanpeca** [comment]
>The "Setup Conda" step when used with mamba seems to take a long time without installing my custom deps only mamba-version to "*" (~6 min). A mamba only setup would probably fix this.

Yep that is possible. Investigating these delays and trying to mitigate them.

**Comment by bollwyvl** [comment]
Provided #126 goes through, this is ready for business, as in https://github.com/conda-incubator/setup-miniconda/issues/76#issuecomment-748717482

The difference here is that `useMamba` would need to get hoisted, earlier, and more forcefully, as no `conda` command would be available. It's also probably _not_ a `constructor`-compatible CLI, but I could be wrong... if not, we'll need to do some updating in a few more places.

**Comment by goanpeca** [comment]
@conda-incubator/setup-miniconda do we want to support micromamba on this action?

**Comment by jaimergp** [comment]
I'd rather not, there's an excellent `setup-micromamba` for those that need that <3

**Comment by goanpeca** [comment]
We can close this one then !
